backend/src/Edna/Setup.hs

```diff
@@ -41,11 +51,51 @@ runEdna config action = do
           }
     runRIO ednaContext action
 
--- If @EDNA_DEBUG_DB@ environment variable is set to @1@
--- or @TRUE@ (case-insensitive), we will run postgres with
--- debug logging.
-askDebugDB :: IO Bool
-askDebugDB = lookupEnv "EDNA_DEBUG_DB" <&> \case
-  Just "1"                       -> True
-  Just (map C.toLower -> "true") -> True
-  _                              -> False
+-- | Dump config to stdout
+dumpConfig :: EdnaConfig -> IO ()
+dumpConfig = B.putStr . Y.encode
+
+-- | Prepare 'EdnaConfig', priority - CLA, env, config file
+prepareConfig :: EdnaOptions -> IO EdnaConfig
+prepareConfig claOptions = do
+  -- use config file if specified or default config otherwise
+  config <- maybe (pure defaultEdnaConfig) Y.decodeFileThrow $
+              eoConfig claOptions
+
+  api <- prepareApiConfig (_ecApi config) claOptions
+  db <- prepareDbConfig (_ecDb config) claOptions
+  logging <- prepareLoggingConfig (_ecLogging config) claOptions
+
+  return $ EdnaConfig api db logging
+
+chooseConfig :: a -> Maybe a -> IO (Maybe a) -> IO a
+chooseConfig _ (Just claOption) _ = return claOption
+chooseConfig configOption Nothing envOption = do
+  env <- envOption
+  case env of
+    Nothing -> return configOption
+    Just x -> return x
+
+prepareApiConfig :: ApiConfig -> EdnaOptions -> IO ApiConfig
+prepareApiConfig apiConfig claOptions = do
+  listen <- chooseConfig (_acListenAddr apiConfig) (eoListen claOptions) apiListenEnv
+  docs <- chooseConfig (_acServeDocs apiConfig) (eoDocs claOptions) serveDocsEnv
+  return $ ApiConfig listen docs
+
+prepareConnection :: ConnString -> IO ConnString
+prepareConnection conn = do
+  envConn <- connectionEnv
+  let connection = case envConn of
+                    Nothing -> conn
+                    Just c -> ConnString $ unConnString conn <> unConnString c
```

Comment:
1. Shouldn't we insert `" "` between `ConnString`s?
2. I guess you can define `Semigroup` and `Monoid` instances for `ConnString`, they seem reasonable to me.

Reply:
I put a space when build hole ConnString from envs
But it'll be better to add instances